rules/private/copy_file_private.bzl

```diff
@@ -19,13 +19,19 @@ cmd.exe (on Windows). '_copy_xfile' marks the resulting file executable,
 '_copy_file' does not.
 """
 
+def _hash_file(file):
+    return str(hash(file.path))
+
 def copy_cmd(ctx, src, dst):
     # Most Windows binaries built with MSVC use a certain argument quoting
     # scheme. Bazel uses that scheme too to quote arguments. However,
     # cmd.exe uses different semantics, so Bazel's quoting is wrong here.
     # To fix that we write the command to a .bat file so no command line
     # quoting or escaping is required.
-    bat = ctx.actions.declare_file(ctx.label.name + "-cmd.bat")
+    # Put a hash of the file name into the name of the generated batch file to
```

Comment:
note: users of the public API wouldn't observe a problem because only one declare_file happens per `copy_file` rule, and so `ctx.label.name` is sufficient to disambiguate.

We hit this in a scenario where a rule uses this library code and creates multiple actions.